compiler/ghc/default.nix

```diff
@@ -185,17 +189,43 @@ let
 
   targetCC = builtins.head toolsForTarget;
 
+  useHadrian = builtins.compareVersions ghc-version "9.4" >= 0;
+
+  inherit ((buildPackages.haskell-nix.cabalProject {
+      compiler-nix-name = "ghc8107";
+      src = haskell-nix.haskellLib.cleanSourceWith {
+        src = buildPackages.srcOnly {
+          name = "hadrian";
+          inherit src;
+          patches = ghc-patches;
+        };
+        subDir = "hadrian";
+      };
+    }).hsPkgs.hadrian.components.exes) hadrian;
+
+  hadrianArgs = "--flavour=${
+        "default"
+          + lib.optionalString (!enableShared) "+no_dynamic_ghc"
+          + lib.optionalString useLLVM "+llvm"
+      } --docs=no-sphinx -j -V";
```

Comment:
Should we document what these flags mean? `-V` especially?
Also where to find the `flavour` combinators?

Reply:
Added comment and replaced `-V` with `--verbose`